app/Http/Controllers/Web/GroupsController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Rogue\Http\Controllers\Web;
+
+use Rogue\Models\Group;
+use Illuminate\Http\Request;
+use Rogue\Http\Controllers\Controller;
+
+class GroupsController extends Controller
+{
+    /**
+     * Create a controller instance.
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
+        $this->middleware('role:admin,staff');
+
+        $this->rules = [
+            'name' => 'required|unique:groups,group_type_id',
+        ];
+    }
+
+    /**
+     * Create a new group.
+     */
+    public function create($groupTypeId)
+    {
+        return view('groups.create')->with(['groupTypeId' => (int) $groupTypeId]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     */
+    public function store(Request $request)
+    {
+        $request->validate($this->rules);
+
+        $group = Group::create($request->all());
```

Comment:
I'm shaky on this, but is it safe to pass the request data wholesale to the create/update calls? (Are we fine since we've defined a `Group::fillable` list of whitelisted values?)

Reply:
Yup, I think the `fillable` handles this concern. This controller is mostly copying by example from other controllers in the repo (although, that shouldn't exactly be the best excuse for assuming it's safe!):

https://github.com/DoSomething/rogue/blob/f660e391e988b84aad379b11066c3d3c9138a366/app/Http/Controllers/Web/ActionsController.php#L71

Reply:
lol but yes, the `fillable` seems like a good safety measure! (Though perhaps worth adding a `group_type_id` validation rule as well? 🤔)